### PR TITLE
Hwinata add genome distribution plot

### LINKED ISSUE #88: Add a plot of distribution of SNVs per cluster for quality control
This plot can be used to look at the distribution of SNVs within each cluster returned by a SRC tool. Plotting the SNVs per cluster is helpful to assess the validity of clusters. We can see if there is any positional bias and the frequency of SNVs at certain genomic regions. 

STGHKGFH000001_multipanelplot_SNVs_genome_distribution_per_cluster

The code to generate these plots is [here](https://github.com/uclahs-cds/project-disease-KidneyTumor-GHLR-000108-SubclonalArchitecture/blob/1a7734ba73553d857516559482036f75c5a9063a/sSRC-prep-and-analysis/analyze_SNVs_in_clones/plot_cluster_SNV_genome_distribution_densityplot.R). 
I used results from the SRC tool DPClust. When I wrote this code, I only had three samples, so I processed each sample separately using functions.

* The first input is a tab-delimited `.txt` file containing the following information and column names:
     | 1st column | 2nd column | 3rd column |
     | --- | --- | --- |
     | chromosome (`chr`) | SNV position for that chromosome (`pos`) | cluster (`cluster`) |
* The second input is a tab-delimited `.txt` file with the following information and column names:
     | 1st column | 2nd column | 3rd column | 4th column |
     | --- | --- | --- | --- |
     |  chromosome (`chr`) | SNV position for that chromosome (`pos`) | cluster (`cluster`) | position along the whole genome (`genomePOS`)
     * This file was generated from a previous step and saved, but this logic is used to generate the genomic positions:
     
          image
     where `vcf.df` is a dataframe of SNVs (must have chromosome and position information) and `chrStart` are the starting positions along the genome of each chromosome. Code to obtain these start positions are included in the script linked above.

### PR DESCRIPTION
**Pull request by whelena**
# Description
Added a QC plot made by @WuSelina for visualizing clone distribution across the genome

## Pipeline Run Results

- Case 1
    - sample:    MSK-AB-0014
    - input tsv: `/hot/software/package/public-R-CancerEvolutionVisualization/development/test_input/multi-sample.tsv`
    - output:    `/hot/software/package/public-R-CancerEvolutionVisualization/development/hwinata-add-genome-distribution-plot/no-defaults`
    - 
# Checklist

- [x] This PR **does *NOT* contain** Protected Health Information [(PHI)](https://ohrpp.research.ucla.edu/hipaa/). A repo may ***need to be deleted*** if such data is uploaded. <br> Disclosing PHI is a ***major problem***[^1] - Even ***a small leak can be costly***[^2].
  
- [x] This PR **does *NOT* contain** germline genetic data[^3], RNA-Seq, DNA methylation, microbiome or other molecular data[^4].

[^1]: [UCLA Health reaches $7.5m settlement over 2015 breach of 4.5m patient records](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m)
[^2]: [The average healthcare data breach costs $2.2 million, despite the majority of breaches releasing fewer than 500 records.](https://www.ponemon.org/local/upload/file/Sixth%20Annual%20Patient%20Privacy%20%26%20Data%20Security%20Report%20FINAL%206.pdf)
[^3]: [Genetic information is considered PHI.](https://www.genome.gov/about-genomics/policy-issues/Privacy#:~:text=In%202013%2C%20as%20required%20by,genetic%20information%20for%20underwriting%20purposes.)
  [Forensic assays can identify patients with as few as 21 SNPs](https://www.sciencedirect.com/science/article/pii/S1525157817305962)
[^4]: [RNA-Seq](https://www.nature.com/articles/ng.2248), [DNA methylation](https://ieeexplore.ieee.org/document/7958619), [microbiome](https://www.pnas.org/doi/pdf/10.1073/pnas.1423854112), or other molecular data can be used to predict genotypes (PHI) and reveal a patient's identity.

- [x] This PR **does *NOT* contain** other non-plain text files, such as: compressed files, images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other output files.

_&emsp; To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example._

- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have set up or verified the `main` branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
  
- [x] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

### CODE REVIEW

**Comment by whelena** [comment]
For chromosome information, I'm currently just readiung from a .tsv file i stuck into `data/chr.info`. I don't know if this is the best approach so any comments is much appreciated.

**Review by WuSelina — APPROVED** [review]
This looks good to me. Thank you for helping me add it here, and great idea adding/defaulting to the genome build for GRCh37 too. 

I think there is a fail on the R-CMD-check regarding documentation warnings though?

**Comment by WuSelina** [comment]
> For chromosome information, I'm currently just readiung from a .tsv file i stuck into `data/chr.info`. I don't know if this is the best approach so any comments is much appreciated.

**I think the way you have it is good.** I am also not sure of what the recommended standard is, though. 

I previously used `get.chr.length()` from the `bedr` package which gives lengths for GRCh38 if specified, but this function does not return 'GC_count' or 'GC_percent' and returns out-of-order chrs (19 & 20 are swapped), so I have been reordering the resulting dataframe just in case:

`# Get chr lengths info`
`chr.len <- get.chr.length(build = 'hg38');`
`# Keep only chrs 1-22 and sex chrs and remove 'chr' prefix`
`chr.len$chr <- gsub('chr', '', chr.len$chr);`
`chr.len <- subset(chr.len, subset = chr %in% c(1:22, 'X', 'Y'));`

`# Reorder the chr info`
`chrom.order <- c(as.character(1:22), 'X', 'Y');`
`# Convert the 'chr' column to a factor with custom levels`
`chr.len$chr <- factor(chr.len$chr, levels = chrom.order);`
`# Sort the dataframe based on the order of the 'chr' column`
`chr.len <- chr.len[order(chr.len$chr), ];`

File: R/create.clone.genome.distribution.scatterplot.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    return(BoutrosLab.plotting.general::create.scatterplot(
+        filename = save.plt,
+        formula = clone.id ~ genome.pos,
+        data = scatter.df,
+        xlab.top.lab = paste0('Total SNVs: ', nsnv),
+        xlab.top.y = 0.5,
+        xlab.label = xlab.label,
+        ylab.label = 'Clone ID',
+        xaxis.lab = chr.info$chr,
+        yaxis.lab = rev(levels(scatter.df$clone.id)),
```
Comment (Faizal-Eeman) [inline R/create.clone.genome.distribution.scatterplot.R:22]: Labels are reversed. This line needs changing.
```suggestion
        yaxis.lab = levels(scatter.df$clone.id),
```

**Comment by whelena** [comment]
@WuSelina Could you double check the density calculation to get the counts? The `density.df$scaled.y` code was not doing what i think it should be doing and giving me really small numbers. The function is under `create.clone.genome.distribution.densityplot.R`. Thanks!

**Comment by whelena** [comment]
R-CMD-check is failing due to missing documentation, which I will be fixing in a separate PR.

**Pull request merged**